src/Grpc.AspNetCore.Server.ClientFactory/ContextPropagationInterceptor.cs

```diff
@@ -129,6 +132,19 @@ public ContextPropagationInterceptor(GrpcContextPropagationOptions options, IHtt
         return response;
     }
 
+    // Automatically dispose state after awaiting the response.
+    private static async Task<TResponse> OnResponseAsync<TResponse>(Task<TResponse> task, IDisposable state)
+    {
+        try
+        {
+            return await task.ConfigureAwait(false);
```

Comment:
Might this be a rare occasion when ContinueWith makes more sense?

Reply:
Why would it be better here? From what I know, async/await is preferred.